### ISSUE TITLE
Understand loss.shape

### ISSUE THREAD

**Issue by BAAAL00** [BlueCl0wn/Robotic-Arm-Controller#3]

**Comment by markorn1612** [comment]
see: https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html

reduction ([str](https://docs.python.org/3/library/stdtypes.html#str), optional) – Specifies the reduction to apply to the output: 'none' | 'mean' | 'sum'. 'none': no reduction will be applied, 'mean': the sum of the output will be divided by the number of elements in the output, 'sum': the output will be summed. Note: size_average and reduce are in the process of being deprecated, and in the meantime, specifying either of those two args will override reduction. Default: 'mean'

We used default: 'mean', so the output is a scalar value, this results in loss.shape being torch.Size([])

If we want the loss for each element, we should use reduction='none', which results in loss.shape being the same as input and target.
